api/turing/cluster/labeller/labeller.go

```diff
@@ -28,9 +32,10 @@ func InitKubernetesLabeller(p, e string) {
 
 // KubernetesLabelsRequest helps to build the Kubernetes labels needed for Kubernetes objects.
 type KubernetesLabelsRequest struct {
-	Stream string
-	Team   string
-	App    string
+	Stream    string
+	Team      string
+	App       string
+	Component string
```

Comment:
For PodDisruptionBudget's LabelSelector to work correctly, I added component label here. @krithika369

Reply:
Hmm.. Adding lots of labels has a cost too (like with metrics scraping, unless we selectively drop some labels downstream). Instead, can we rely on the default `app` label added by the k8s Deployment controller, the same way that the TopologySpreadConstraints work? https://github.com/caraml-dev/turing/blob/main/api/turing/cluster/knative_service.go#L165